.azure-pipelines/MacOS-CI.yml

```diff
@@ -40,76 +40,36 @@ jobs:
       versionSpec: '$(python.version)'
 
   - script: |
-      # Install Protobuf from source
       export NUM_CORES=`sysctl -n hw.logicalcpu`
       if [ '$(onnx_debug)' == '1' ]; then
-        export DEBUG=1
         source workflow_scripts/protobuf/build_protobuf_unix.sh $NUM_CORES $(pwd)/protobuf/protobuf_install Debug
       else
         source workflow_scripts/protobuf/build_protobuf_unix.sh $NUM_CORES $(pwd)/protobuf/protobuf_install
       fi
-
+    displayName: 'Install Protobuf from source'
+  - script: |
       git submodule update --init --recursive
-      python -m pip install -q --upgrade pip
-      python -m pip install -q -r requirements-release.txt
-
-      if [ '$(onnx_ml)' == '1' ]; then
-        export ONNX_ML=1
-      fi
-      export ONNX_BUILD_TESTS=1
-      export CMAKE_ARGS="-DONNX_WERROR=ON"
-      if [ '$(onnx_lite)' == '1' ]; then
-        export CMAKE_ARGS="${CMAKE_ARGS} -DONNX_USE_LITE_PROTO=ON"
-      fi
-      export ONNX_NAMESPACE=ONNX_NAMESPACE_FOO_BAR_FOR_CI
-      python setup.py -q install
-    displayName: 'Install dependencies and ONNX'
+    displayName: 'Load all submodules'
 
   - script: |
-      # check auto-gen files up-to-date
-      python onnx/defs/gen_doc.py
-      python onnx/gen_proto.py -l
-      python onnx/gen_proto.py -l --ml
-      python onnx/backend/test/stat_coverage.py
-
-      git status
-      git diff --exit-code  -- . ':(exclude)onnx/onnx-data.proto' ':(exclude)onnx/onnx-data.proto3'
-      if [ $? -ne 0 ]; then
-        echo "git diff returned failures"
-        exit 1
-      fi
-
-      # Do not hardcode onnx's namespace in the c++ source code, so that
-      # other libraries who statically link with onnx can hide onnx symbols
-      # in a private namespace.
-      ! grep -R --include='*.cc' --include='*.h' 'namespace onnx' .
-      ! grep -R --include='*.cc' --include='*.h' 'onnx::' .
+      pip install nox
+    displayName: 'Install nox'
 
-      pytest
-      if [ $? -ne 0 ]; then
-        echo "pytest failed"
-        exit 1
-      fi
-
-      # onnx c++ API tests
-      export LD_LIBRARY_PATH="./.setuptools-cmake-build/:$LD_LIBRARY_PATH"
-      ./.setuptools-cmake-build/onnx_gtests
-      if [ $? -ne 0 ]; then
-        echo "onnx_gtests failed"
-        exit 1
-      fi
+  - script: |
+      nox --forcecolor --no-venv -- -v --cov=onnx --cov-report=xml --cov-append --cov-branch
+    env:
+      DEBUG: $(onnx_debug)
+      ONNX_ML: $(onnx_ml)
+      ${{ if eq(variables['onnx_lite'], true) }}:
+        CMAKE_ARGS: "-DONNX_WERROR=ON -DONNX_USE_LITE_PROTO=ON"
```

Comment:
@jcwchen  Do we need `DONNX_USE_PROTOBUF_SHARED_LIBS` for macos too? I am getting "  Protobuf compiler not found" with the original config (I think)

https://dev.azure.com/onnx-pipelines/onnx/_build/results?buildId=36838&view=logs&j=6249d9e9-4a84-5ec2-7b7b-c0dee11b2acc&t=c4813265-867c-5f6d-4ed7-35346f517ab8&l=2914

Reply:
Not really, because MacOS.yml buils Protobuf from source by `source workflow_scripts/protobuf/build_protobuf_unix.sh $NUM_CORES $(pwd)/protobuf/protobuf_install` so it uses static Protobuf instead.